internal/controller/progressive/progressive.go

```diff
@@ -494,13 +492,18 @@ func startUpgradeProcess(
 	if err != nil {
 		return newUpgradingChildDef, false, err
 	}
+	// reset the Status to reflect our new Upgrading child
+	err = rolloutObject.ResetUpgradingChildStatus(newUpgradingChildDef)
+	if err != nil {
+		return newUpgradingChildDef, false, fmt.Errorf("processing upgrading child, failed to reset the upgrading child status for child %s/%s", newUpgradingChildDef.GetNamespace(), newUpgradingChildDef.GetName())
+	}
 
 	requeue, err = controller.ProcessUpgradingChildPreUpgrade(ctx, rolloutObject, newUpgradingChildDef, c)
 	if err != nil {
-		return nil, false, err
+		return newUpgradingChildDef, false, err
 	}
 	if requeue {
-		return nil, true, nil
+		return newUpgradingChildDef, true, nil
```

Comment:
We were being inconsistent about returning this in the cases where it wasn't really being used on the other side. In these 2 cases we weren't returning it, but in other cases (like below) we were.